### PR TITLE
Fix Azure integration cleanup

### PR DESCRIPTION
**Pull request by flin-8**
For Azure, the `Apply` and `Destroy` commands weren't running from the same directory, `Destroy` didn't have access to the state file, resulting in it not cleaning up. AWS was doing this by setting `OriginalPackageDirectoryPath`. I'm doing the same for Azure now.

We were going to use the `-state=xxx` argument to ensure the state was shared, but unfortunately there's a small bug/issue with the [ApplyCommand](https://github.com/OctopusDeploy/Sashimi.Terraform/blob/65b05befca76ef3b67f7f9d4df892064c2dab6a8/source/Calamari.Terraform/ApplyCommand.cs#L36), we also need to pass the -state=xxx argument to the output command for it to get the correct results. 

Reviewing commit by commit is easiest

### CODE REVIEW

File: source/Sashimi.Terraform.Tests/ActionHandlersFixture.cs [hunk 1]
```diff
@@ -327,41 +331,37 @@ void PopulateVariables(TestActionHandlerContext<Program> _)
                 _.Variables.Add("random", random);
                 _.Variables.Add(TerraformSpecialVariables.Action.Terraform.VarFiles, "example.tfvars");
                 _.Variables.Add(TerraformSpecialVariables.Action.Terraform.AzureManagedAccount, Boolean.TrueString);
+                _.Variables.Add(KnownVariables.OriginalPackageDirectoryPath, temporaryFolder.DirectoryPath);
```
Comment (flin-8) [inline source/Sashimi.Terraform.Tests/ActionHandlersFixture.cs:357]: This is the actual fix

**Review by johnsimons — APPROVED** [review]
Seems reasonable, but shouldn't we just fix the bug you found?

**Comment by flin-8** [comment]
> Seems reasonable, but shouldn't we just fix the bug you found?

We would need to parse the additional arguments to separate out the `-state=xxx` argument, which is non-trivial. I will log a bug with QRF

File: source/Sashimi.Terraform.Tests/ActionHandlersFixture.cs [hunk 2]
```diff
@@ -396,72 +396,52 @@ void PopulateVariables(TestActionHandlerContext<Program> _)
...
+            var output = ExecuteAndReturnResult(typeof(TerraformPlanActionHandler), PopulateVariables, "AWS");
+            output.OutputVariables.ContainsKey("TerraformPlanOutput").Should().BeTrue();
 
-                string fileData;
-                using (var client = new HttpClient())
-                {
-                    fileData = await client.GetStringAsync($"https://{bucketName}.s3.amazonaws.com/test.txt").ConfigureAwait(false);
-                }
+            output = ExecuteAndReturnResult(typeof(TerraformApplyActionHandler), PopulateVariables, "AWS");
+            output.OutputVariables.ContainsKey("TerraformValueOutputs[url]").Should().BeTrue();
```
Comment (flin-8) [inline source/Sashimi.Terraform.Tests/ActionHandlersFixture.cs:406]: These output variables disappear if you add `-state=xxx` as additional command line arguments like on [line 431](https://github.com/OctopusDeploy/Sashimi.Terraform/pull/2/files#diff-5173d5f99a2fe9258c85d0b0005adc90R431)

**Pull request merged**